### ISSUE TITLE
getLocations doesn't return all locations in 3.0.0-alpha.48

### ISSUE THREAD

**Issue by angel1st** [mauron85/cordova-plugin-background-geolocation#520]
## Your Environment

* Plugin version:
3.0.0-alpha.48
* Platform: Android
* OS version:
8.0.0 / 7.0.0
* Device manufacturer and model:
S8 / Redmi Note4
* Cordova version (`cordova -v`):
8.1.2
* Cordova platform version (`cordova platform ls`):
android 7.1.0
ios 4.5.5
* Plugin configuration options: 
```
{
       locationProvider: BackgroundGeolocation.ACTIVITY_PROVIDER,
       desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
       stationaryRadius: distance,
       distanceFilter: distance,
       notificationTitle: "Locator is tracking",
       notificationText: "Locator is enabled",
       debug: false,
       interval: 10000,
       fastestInterval: 5000,
       activitiesInterval: 10000,
       stopOnTerminate: true,
       startOnBoot: false,
       startForeground: true,
       maxLocations: 100000
}
```
* Link to your project:

## Context

I am invoking `BackgroundGeolocation.getLocations` to get all stored locations or at least the last 100k.

## Expected Behavior

I was expecting to receive all locations stored (or at least the last 100k), instead, I have just few, but certainly not all. I compared **alpha.48** to **alpha.47** where all is working just fine and I can retrieve all locations stored for the certain period.

## Actual Behavior

I have received few locations but they are not all. To compare - I have the app compiled with 47. In this case I have received approx 60 locations for ~10 mins. In 48, for the same timespan and going through the same route, I have received just 6 locations.
Please note, I kept all other options the same, only plugin changed.

BTW - I noticed one visual difference - until 47 version, the plugin created `dummy` account into device's account place. Since 48, the account now is named after the app.
 
## Steps to Reproduce

1. Compile with 47 and make some movements (I walked around approx 10 mins)
1. Invoke `BackgroundGeolocation.getLocations` and see how many locations have been returned from the plugin
1. Compile with 48 and follow the same route and time duration
1. Invoke `BackgroundGeolocation.getLocations` and check the number of returned locations. In my case they were 10 times less.

**Comment by mauron85** [comment]
Any chance you've used deleteLocation(s)?

**Comment by angel1st** [comment]
No actually - as I said, the only change between two builds is plugin version. 
In fact I don't have `deleteLocation` in my entire code.

**Comment by mauron85** [comment]
Just to make sure. All locations (100K) are there, when you downgrade from 48 to 47?

**Comment by angel1st** [comment]
I cannot downgrade from 48 to 47. Instead, I uninstall one, then install another and repeat the route. So basically, I cannot confirm the locations are collected into the local db or not.

**Comment by mauron85** [comment]
I've run test side by side with two apps one using alpha.47 and the other one alpha.48 with following config options:

```
        BackgroundGeolocation.configure({
            locationProvider: BackgroundGeolocation.RAW_PROVIDER,
            desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
            stationaryRadius: 1,
            distanceFilter: 1,
            notificationTitle: "Locator is tracking",
            notificationText: "Locator is enabled",
            debug: false,
            interval: 1000,
            fastestInterval: 1000,
            stopOnTerminate: true,
            startOnBoot: false,
            startForeground: true,
            maxLocations: 100000            
        });
```

And number of locations returned by `getLocations` were same. I can test with ACTIVITY_PROVIDER on real device, but don't expect any difference either (as there was no change in Activity Provider in alpha.48)

**Comment by angel1st** [comment]
@mauron85 - thanks for the feedback. Today I did following, just to confirm the issue. I have one project in development, in which I use the plugin. So far - all 3.0.0 alpha versions (I believe I started from 42) were working just fine. Let me describe the exact steps:

### Prerequisites
Config file - as described in the issue template, the only missing piece is `distance = 15`

### alpha.47
1. Uninstall the app from the device
1. > cordova plugin rm cordova-plugin-mauron85-background-geolocation
1. > cordova platform rm android
1. > cordova platform rm ios
1. > cordova plugin add cordova-plugin-mauron85-background-geolocation@3.0.0-alpha.47
1. > cordova platform add android
1. > cordova platform add ios
1. > cordova build
1. Install the app on device & start it
1. Right after the start (please note, the device is basically still) if I execute `getLocations` I can see 2 locations already created

### alpha.48
1. Repeat 1, 2, 3, 4
1. > cordova plugin add cordova-plugin-mauron85-background-geolocation@3.0.0-alpha.48
1. Repeat the remaining steps - now `getLocations` returns empty array.

Please note, the **only** difference between two build is the plugin version. This is tested on two real devices as mentioned above.

**Note** - one visual difference between two plugin releases is as follow - until 47, the plugin creates account named **dummy** into device accounts. Since 48, the account is now having the app name. I don't know if this one is somehow related but I believe it is worthwhile to be mentioned.

The bottom line is - there is certainly an issue and it would be nice if it could be fixed.

**Comment by mauron85** [comment]
Ok, I don't think there is any issue with alpha.48. Locations count will be different even if you run same version of the plugin and clear data and restart app. It cannot be exactly same everytime you run, because conditions are different every single time (strength of the gps signal or gps/network provider, etc...)

**Comment by angel1st** [comment]
In the case described above, the conditions are identical as much as possible. Also, I still claim, the overall number of locations recorded by 48 are in **times** less compared to previous versions.
If you look at the initial issue template, the difference is **10 times** - all recorded on the same device going through the same route. I would agree if the difference is 3 to 5 percents, considering other conditions as signal strength and etc., but 10000% seems quite unusual.

**Comment by angel1st** [comment]
Here are some real field data from today. I just compiled the app with **49** and even adjusted `interval`, `fastestInterval` and `activtiesInterval` to the above values from your config.
These are the values as returned from Chrome console (run time data):

```
activitiesInterval: 1000
debug: false
desiredAccuracy: 0
distanceFilter: 15
fastestInterval: 500
httpHeaders: {}
interval: 1000
locationProvider: 1
maxLocations: 100000
notificationIconColor: ""
notificationIconLarge: ""
notificationIconSmall: ""
notificationText: "--"
notificationTitle: "----"
notificationsEnabled: true
startForeground: true
startOnBoot: false
stationaryRadius: 15
stopOnStillActivity: true
stopOnTerminate: true
syncThreshold: 100
syncUrl: ""
url: "https://api.example.com/location"
```

I have just returned from about 60 minute walk with my S8 with the app. The walk took place between 14:00 and 15:00 EET (GMT+2).
Then when I was back, I connected the device and invoked `getLocations`. Please review the enclosed file - [20181203-2.txt](https://github.com/mauron85/cordova-plugin-background-geolocation/files/2639604/20181203-2.txt). You won't see a single location saved between the walking timespan (14:00 and 15:00). You will see some locations saved **after** the walk. I can confirm however, the app was running and the marker was moving along my walk.

**Note** - I have removed `lat` and `lon` from the saved locations for privacy concerns. All other data are kept untouched as returned from the method.

Let me know shall you need any further inputs to tackle this issue.

**Comment by angel1st** [comment]
@mauron85 - to complete the feedback and tests. 
Today, I took **exactly** the same route as yesterday by foot. Now I, have my S8 / 8.0.0 with app compiled with **47**. For the very same **route and timespan** (approx. one hour) now I have **550** locations returned with `getLocations`. Let me know if you want me to send you this file as well.
In addition, I also brought one more android device - Redmi Note 4 / 7.0.0 and I can confirm on it the very same results - `getLocations` doesn't return locations from app using **49** and works flawlessly with **47**.
Again - the only difference between two app builds is plugin version, all other is completely identical.

This are the steps I followed to install each app:
1. Uninstall current version - to make sure all data are removed.
1. Install new version and start walking.

I would appreciate your feedback.

**Comment by mauron85** [comment]
Will do some more test with two of my Android devices.

**Comment by mauron85** [comment]
In mean time, can you check if whole route was logged? (eg. render locations on the map) and also check plugin logs. For example print all logs from `ActivityRecognitionLocationProvider`. You can do it in chrome `chrome://inspect` developer's console.

```
BackgroundGeolocation.getLogEntries(10000, 0, 'DEBUG', entries => console.table(entries.filter(entry => entry.logger === 'com.marianhello.bgloc.provider.ActivityRecognitionLocationProvider'), ['id','timestamp','message']));
```

**Comment by angel1st** [comment]
@mauron85 - I haven't able to provide above the last days. Do you still need it? To clarify - do you need this for 49 build only?
Anyway - do you have any progress resolving the issue? I can see version **alpha.50** has been released in the meantime....
